eox_nelp/middleware.py

```diff
@@ -0,0 +1,115 @@
+"""Middleware file.
+
+Required NELP middlewares that allow to customize edx-platform.
+
+classes:
+    ExtendedProfileFieldsMiddleware: Set extended_profile_fields in registration form.
+"""
+from eox_nelp.edxapp_wrapper.site_configuration import configuration_helpers
+from eox_nelp.edxapp_wrapper.user_authn import RegistrationFormFactory
+from eox_nelp.edxapp_wrapper.user_api import accounts
+
+
+# Copy default values from https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp
+# /openedx/core/djangoapps/user_authn/views/registration_form.py#L308
+DEFAULT_EXTRA_FIELDS = [
+    "confirm_email",
+    "first_name",
+    "last_name",
+    "city",
+    "state",
+    "country",
+    "gender",
+    "year_of_birth",
+    "level_of_education",
+    "company",
+    "job_title",
+    "title",
+    "mailing_address",
+    "goals",
+    "honor_code",
+    "terms_of_service",
+    "profession",
+    "specialty",
+]
+
+
+class ExtendedProfileFieldsMiddleware:
+    """This middleware class will update the registration form factory extra fields
+    based on the tenant setting "extended_profile_fields", therefore the new values will be
+    settable from the register form in /register, from /account/settings and finally from
+    SAML IDP response.
+
+    Configuration example, add the following in your tenant config:
+
+        extended_profile_fields = [
+            "hobby",
+            "sport",
+            "movie"
+        ]
+
+        # if you add this in general setting the default values will be deleted
+        REGISTRATION_EXTRA_FIELDS = {
+            'hobby': 'required',
+            'sport': 'required',
+            'movie': 'optional'
+        }
+
+        # if you want to add specific options to the  new fields
+
+        EXTRA_FIELD_OPTIONS = {
+            'sport': ['soccer', 'football', 'tennis'],
+            'movie': ['Harry Potter', 'Karate kid']
+        }
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # The following line resets to the default values, to avoid tenants overlapping.
```

Comment:
What do you mean by the tenant overlapping. So it will always reset to this list?
https://github.com/eduNEXT/eox-nelp/pull/12/files#diff-66a15828789bf584965751e47deaa844139ec753886961cc4c1450c06e733c70R15

Reply:
An example, tenant A, with field movie, will add that field but if the next request comes from tenant B and that tenant doesn't have extra fields without that line the tenant B will show the movie field, so yes it will always reset the list to the original value

Reply:
That is interesting. so in that condition, I can't have tenants with fewer fields. But I think that that is not problem... who should remove REGISTER_FIELDS xD.

Reply:
This modifies EXTRA_FIELDS inside the class RegisterFormFactory,  that is a constant value therefore you will never have less fields. https://github.com/eduNEXT/eox-nelp/pull/12/files#diff-66a15828789bf584965751e47deaa844139ec753886961cc4c1450c06e733c70R15  previous line is a identical copy  of EXTRA_FIELDS you can find that values here https://github.com/eduNEXT/edunext-platform/blob/ednx-release/mango.master.nelp/openedx/core/djangoapps/user_authn/views/registration_form.py#L308